ONSdigital/eq-author-app
# EAR-1737-Dynamic-Answers

### What is the context of this PR?

> This PR allows Authors to be able to use an undefined number of checkbox answer labels to populate another set (subsequent) of radio answer labels as the respondent needs see previously selected answers in order to answer a follow on question. 

### How to review

1. Create a checkbox answer
2. Create a radio answer

> - [ ] Check dynamic option toggle switch appears for a radio answer 


4. Turn the dynamic option toggle switch on 

> - [ ] Check when the dynamic option toggle switch is on the dynamic answer content picker button displays and the label and description fields disappear.


5. Click the dynamic answer content picker button and select a checkbox answer in the modal and click confirm

> - [ ]  Check the correct checkbox answer is displayed in the content picker modal
> - [ ]  Check when the answer is clicked in the modal the background for the selected answer turns blue
> - [ ]  Check the correct checkbox answer is displayed in the  dynamic answer content picker button after clicking confirm

Test that it works correctly with multiple checkbox answers in the questionnaire - repeating the steps above

**Validation** 

- [ ] Check the validation message "Answer required" displays under the  dynamic answer content picker button when an answer hasn't been selected
- [ ]  Check the validation message "The selected answer has been deleted" displays under the  dynamic answer content picker button when the checkbox answer selected has been deleted 
- [ ] Check the validation message "Answer must be from a previous question" displays under the  dynamic answer content picker button when the checkbox answer selected has moved to a position after the current page

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
3. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
4. - [ ] Click the **Merge** button on this pull request
6. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
7. - [ ] Move the Jira ticket for this task into the next stage of the process
